File: apps/dotnet/Functions/EventHubLagProcessorFunction.cs
using Microsoft.Azure.Functions.Worker;
using Microsoft.Extensions.Logging;

namespace AzureFunctionsGuide.Functions;

/// <summary>
/// Event Hub message processor — processes events from the events hub.
/// Binding: EventHub trigger on "events" with EventHubConnection connection.
/// </summary>
public class EventHubLagProcessorFunction
{
    private readonly ILogger<EventHubLagProcessorFunction> _logger;

    public EventHubLagProcessorFunction(ILogger<EventHubLagProcessorFunction> logger)
    {
        _logger = logger;
    }

    [Function("eventhubLagProcessor")]
    public void Run(
        [EventHubTrigger("events", Connection = "EventHubConnection")] string[] messages)
    {
        _logger.LogInformation("EventHub batch received: {Count} messages", messages.Length);

        foreach (var message in messages)
        {
            _logger.LogInformation("EventHub message: {Message}", message);
        }
    }
}

File: apps/java/src/main/java/com/functions/EventHubLagProcessorFunction.java
package com.functions;

import com.microsoft.azure.functions.*;
import com.microsoft.azure.functions.annotation.*;

/**
 * EventHub trigger — processes events from telemetry-events hub.
 */
public class EventHubLagProcessorFunction {

    @FunctionName("eventhubLagProcessor")
    public void run(
            @EventHubTrigger(
                name = "event",
                eventHubName = "telemetry-events",
                connection = "EventHubConnection",
                consumerGroup = "$Default",
                cardinality = Cardinality.ONE)
            String event,
            final ExecutionContext context) {

        context.getLogger().info("EventHub event received: " + event);
    }
}
